### ISSUE TITLE
Performance difference between Sequential and ConcurrentTaskRunner for tasks with synchronous network IO

### ISSUE THREAD

**Issue by AT-SyT** [PrefectHQ/prefect#6468]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

There is a difference in performance between running a SequentialTaskRunner and ConcurrentTaskRunner with synchronous tasks which perform heavy network IO when using synchronous tasks.

I am running into performance issues when downloading large files (via httpx) from an internal API. Those issues only arise when I use a ConcurrentTaskRunner with synchronous tasks. If I use the SequentialTaskRunner or only async tasks the performance is fine. 
Namely the throughput of the API download should be around 800Mbps which is about the limit of the connection and the API. This is the case for the SequentialTaskRunner. If I use the ConcurrentTaskRunner then the speeds fluctuate and I am between 1 and about 25Mbps. 

The issue goes away if I change the download task to use the async functionality of httpx and make the flow and the tasks async. See attached screenshots for the comparisons

The reproduction code includes all four variations in a minimal example to reproduce the issue by downloading a 1GB file from a public url. 

![Screenshot 2022-08-18 at 18 46 01](https://user-images.githubusercontent.com/12372752/185450310-7d9e7360-6b8f-4325-b769-48eb2bd16d8c.png)
![Screenshot 2022-08-18 at 18 46 33](https://user-images.githubusercontent.com/12372752/185450318-38d055f7-e1d2-40c7-97ae-fe8c76cb4f64.png)
![Screenshot 2022-08-18 at 18 46 43](https://user-images.githubusercontent.com/12372752/185450319-fee8d3f8-e475-4e30-a540-1cc8dd68e86e.png)
![Screenshot 2022-08-18 at 18 46 54](https://user-images.githubusercontent.com/12372752/185450320-73ec46dc-82b8-4f2f-b4a9-3a07c18ef076.png)
![Screenshot 2022-08-18 at 18 47 08](https://user-images.githubusercontent.com/12372752/185450322-74ad0e6a-d547-4a1e-b93b-06c36f654f38.png)

### Reproduction

```python
import io
import time

import httpx
from prefect import flow, task, get_run_logger
from prefect.task_runners import ConcurrentTaskRunner, SequentialTaskRunner


async def a_download_file(base_url: str, file: str, target):
    client = httpx.AsyncClient(
        base_url=base_url,
        http2=True)
    async with client.stream('GET', file) as resp:
        async for data in resp.aiter_bytes():
            target.write(data)


def download_file(base_url: str, file: str, target):
    client = httpx.Client(
        base_url=base_url,
        http2=True)
    with client.stream('GET', file) as resp:
        for data in resp.iter_bytes():
            target.write(data)


@task()
def download(url, file):
    logger = get_run_logger()
    logger.info(f'Sync Version')
    logger.info(f'Starting to download {url}/{file}')
    with io.BytesIO() as target:
        start = time.time()
        download_file(
            base_url=url,
            file=file,
            target=target,
        )
        end = time.time()
        logger.info(f"Download took {end - start}s")


@task()
async def a_download(url, file):
    logger = get_run_logger()
    logger.info(f'Async Version')
    logger.info(f'Starting to download {url}/{file}')
    with io.BytesIO() as target:
        start = time.time()
        await a_download_file(
            base_url=url,
            file=file,
            target=target,
        )
        end = time.time()
        logger.info(f"Download took {end - start}s")


@flow(task_runner=SequentialTaskRunner())
def download_sequential():
    logger = get_run_logger()
    logger.info("Synchronous Sequential")
    download.submit("https://speed.hetzner.de", "1GB.bin")


@flow(task_runner=ConcurrentTaskRunner())
def download_concurrent():
    logger = get_run_logger()
    logger.info("Synchronous Concurrent")
    download.submit("https://speed.hetzner.de", "1GB.bin")


@flow(task_runner=SequentialTaskRunner())
async def a_download_sequential():
    logger = get_run_logger()
    logger.info("Asynchronous Sequential")
    await a_download.submit("https://speed.hetzner.de", "1GB.bin")


@flow(task_runner=ConcurrentTaskRunner())
async def a_download_concurrent():
    logger = get_run_logger()
    logger.info("Asynchronous Concurrent")
    await a_download.submit("https://speed.hetzner.de", "1GB.bin")
```

### Error

```

# Copy complete stack trace and error message here, including log output if applicable.

```

### Versions

```

Version:             2.0.3
API version:         0.8.0
Python version:      3.9.5
Git commit:          2f1cf4ac
Built:               Fri, Aug 5, 2022 3:57 PM
OS/Arch:             darwin/x86_64
Profile:             cloud
Server type:         hosted


httpx:                0.22.0
```

### Additional context

The Flows are running inside an Agent that is hosted on an RHEL 8.4 VM with podman using the prefect base image. 
Nothing else was running while performing the test runs

**Comment by madkinsz** [comment]
I believe this is related to a known performance issue with the concurrent task runner described in #6239, but it could be something else. Thanks for the great writeup!

**Comment by AT-SyT** [comment]
I looked a bit into the linked issue and it seems similar. Although I am not comfortable enough with async python as I just recently started to dive into this area. So I am unfortunately not able to help more at the moment.

Attached are screenshots of the cpu usage for the sequential task runner(first small and short spike) followed by a long and high utilization with the concurrent task runner. 
![MicrosoftTeams-image (6)](https://user-images.githubusercontent.com/12372752/185919464-20b4542f-9fa6-4d46-b2c1-2a3e80a1a68c.png)

And in the following image one can see the impact on the bandwidth for the download with the concurrent task runner (which should be quite stable somewhere around 600Mbps)
![MicrosoftTeams-image (7)](https://user-images.githubusercontent.com/12372752/185919838-8866c408-2f63-43c7-b57b-fc47dd416064.png)

**Comment by AT-SyT** [comment]
I looked a bit into the linked issue and it seems similar. Although I am not comfortable enough with async python as I just recently started to dive into this area. So I am unfortunately not able to help more at the moment.

Attached are screenshots of the cpu usage for the sequential task runner(first small and short spike) followed by a long and high utilization with the concurrent task runner.
MicrosoftTeams-image (6)

And in the following image one can see the impact on the bandwidth for the download with the concurrent task runner (which should be quite stable somewhere around 600Mbps)
MicrosoftTeams-image (7)
